docs/trex_format.md

```diff
@@ -0,0 +1,100 @@
+     
+```
+
+### In JavaScript code
+
+To reference the workbook formatting in your JavaScript code, use the enum directly. For example, to add the formatting used in the worksheet title, you could use the JQuery method (`.addClass`) to add the `tableau.ClassNameKey.WorksheetTitle` to an element on your page. The following example applies the formatting to a heading with the id `someTitle`.
+
+```javascript
+
+$('#someTitle').addClass(tableau.ClassNameKey.WorksheetTitle);
+
+```
+
+You could also create a JavaScript function that has `ClassNameKey` as an parameter, and use that function to process and apply the formatting in some way.
+
+```javascript
+
+myFormattingFunction(tableau.ClassNameKey.WorksheetTitle); 
+
+// does something with the worksheet title
+
+```
+
+## Access information about the formatting styles in the workbook
+
+You can access the formatting used in a Tableau workbook from the `tableau.extensions.environment.workbookFormatting` property. The `workbookFormatting` property, `formattingSheets` contains an array of CSS properties for the workbook, organized by `ClassNameKey`.  For example, if you wanted to view the formatting styles available using the Chrome DevTools, you could add the following JavaScript code and then view the results in the Console window by navigating the array. 
+
+```javascript
+
+if (tableau.extensions.environment.workbookFormatting) {
+  console.log(tableau.extensions.environment.workbookFormatting.formattingSheets);
+};
+
+```
+
+Or to just print out the formatting information in the Console window you could do something like the following:
+
+```javascript
+
+let formattingSheets = tableau.extensions.environment.workbookFormatting.formattingSheets;
+formattingSheets.forEach(function (formattingSheet) {
+    console.log("The formatting sheet is " + formattingSheet.classNameKey + " " + JSON.stringify(formattingSheet.cssProperties));
+  });
+
+```
+
+## Set an event listener on workbook format changes
+
+You can set an event listener on changes to the formatting in the workbook. The new event type is `WorkbookFormattingChanged`. The `WorkbookFormattingChanged` event is triggered whenever the workbook formatting is changed in Tableau authoring mode. This includes changes in the font, the font size, whether it is bold, italic, or underlined, and changes in color. 
+
+```javascript
+
+dashboard.addEventListener(tableau.TableauEventType.WorkbookFormattingChanged, (event) => {
+  console.log(event.formatting);
+  // take some other actions based on the change 
+});
+
+```
+
+For more information about using event listeners, see [Events and Event Handling]({{site.baseurl}}/docs/trex_events.html).
+
+## What's Next
+
+* To see a working sample dashboard extension that uses workbook formatting, see the [Formatting](https://github.com/tableau/extensions-api/tree/main/Samples/Formatting){:target="_blank"} sample in the Samples folder.
```

Comment:
For Formatting Sample section, do we also want to reference the one in Samples-Typescript too (https://github.com/tableau/extensions-api/tree/main/Samples-Typescript/Formatting)?

Reply:
@b-carver Yes. I didn't know about that one. I'll add it. Thanks!